File: Serene/Serene.Script/Northwind/Category/CategoryDialog.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [IdProperty("CategoryID"), NameProperty("CategoryName")]
    [FormKey("Northwind.Category"), LocalTextPrefix("Northwind.Category"), Service("Northwind/Category")]
    [Imported]
    public class CategoryDialog : EntityDialog<CategoryRow>
    {
        protected override IEnumerable<System.Tuple<string, string>> GetLanguages()
        {
            return LanguageList.GetValue();
        }
    }
}

File: Serene/Serene.AspNetCore/Modules/Northwind/Category/CategoryDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class CategoryDialog extends Serenity.EntityDialog<CategoryRow, any> {
        protected getFormKey() { return CategoryForm.formKey; }
        protected getIdProperty() { return CategoryRow.idProperty; }
        protected getLocalTextPrefix() { return CategoryRow.localTextPrefix; }
        protected getNameProperty() { return CategoryRow.nameProperty; }
        protected getService() { return CategoryService.baseUrl; }

        protected form = new CategoryForm(this.idPrefix);

        protected getLanguages() {
            return LanguageList.getValue();
        }
    }
}